### ISSUE TITLE
Enable to skip pruning by specified interval for every pruning class.

### ISSUE THREAD

**Issue by keisuke-umezawa** [optuna/optuna#1136]
## Motivation

We want to specify interval to skip pruning, so that we can fasten iterations.
In most cases,  we do not need to run pruning after each iteration, but currently we can not avoid it.

## Description

Enable to skip pruning by specified interval for the certain PruningCallback classes.

- PruningCallback
    - [x] [LightGBMPruningCallback](https://github.com/optuna/optuna/blob/master/optuna/integration/lightgbm.py)
    - [ ] [XGBoostPruningCallback](https://github.com/optuna/optuna/blob/master/optuna/integration/xgboost.py)

**Comment by VladSkripniuk** [comment]
`TensorFlowPruningHook` skips pruning based on `run_every_steps` argument. `ChainerPruningExtension` skips pruning based on `pruner_trigger` argument.
I think we need a discussion to decide whether we need to reformat these two classes to add `interval` argument.

**Comment by toshihikoyanase** [comment]
I think we have an alternative approach to skip pruning. How about adding a method like `BasePruner.is_target_step(step: int) -> bool` so that the callbacks can query the timing of pruning to the pruners?

This is because the pruners have logic to skip pruning. If we add the `interval` argument to the callbacks, users may need to take care of the relationship between the logic and the `interval` argument.

For example, `MedianPruner` has the `n_interval_steps` argument to specify the interval in number of steps between the pruning checks. And `SuccessiveHalvingPruner` determines the pruning steps based on the arguments.

It may be confusing if users use both the `interval` argument and `SuccessiveHalvingPruner` because the pruning interval of `SuccessiveHalvingPruner` increases an exponential manner. So, IMO, the `BasePruner.is_target_step` method will improve usability.

I'm not fully sure if it covers actual use cases. So, if you have any comments and questions, please let me know.

c.f., https://github.com/optuna/optuna/pull/1161#issuecomment-625328649

**Comment by keisuke-umezawa** [comment]
> This is because the pruners have logic to skip pruning. If we add the interval argument to the callbacks, users may need to take care of the relationship between the logic and the interval argument.

That is true. The `interval` should be managed by pruning class.

**Comment by keisuke-umezawa** [comment]
@VladSkripniuk cc: @toshihikoyanase @hvy 

Sorry for being late for updates.

I discussed it with the other developers, and we concluded it as follows:

1. The intervals for reporting and pruning should be configured separately.
2. Pruners should have the information of pruning interval, whereas callbacks should have the information of reporting interval.

Currently, I am implementing the above feature for lightgbm callback in #2490 .
